From 66a1b650896610297c6c8533a45d4cf37a844d21 Mon Sep 17 00:00:00 2001
Message-ID: <66a1b650896610297c6c8533a45d4cf37a844d21.1729453324.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1729453324.git.sharlatanus@gmail.com>
References: <cover.1729453324.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 7 Oct 2024 21:17:37 +0100
Subject: [PATCH 19/42] gnu: python-radio-beam: Update to 0.3.8.

* gnu/packages/astronomy.scm (python-radio-beam): Update to 0.3.8.
[arguments] <#:tests>: Disable, as newer version of NumPy is required.
[native-inputs]: Add python-setuptools and python-wheel. Place before
'propagated-inputs'.

Change-Id: Idca5458b16a70b2ab07d0f30576e0f945b5f0ea0
---
 gnu/packages/astronomy.scm | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index a1cac1ee3b..8929809e3a 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -5905,7 +5905,7 @@ (define-public python-rad
 (define-public python-radio-beam
   (package
     (name "python-radio-beam")
-    (version "0.3.7")
+    (version "0.3.8")
     (source
      (origin
        (method url-fetch)
@@ -5913,13 +5913,24 @@ (define-public python-radio-beam
        (sha256
         (base32 "0dg6vqdhmzh47awdkkcbf455gw6if2qwxyhcqbq2dkhbwsx680gc"))))
     (build-system pyproject-build-system)
+    (arguments
+     ;; See <https://github.com/radio-astro-tools/radio-beam/issues/129>.
+     ;; E   astropy.units.core.UnitScaleError: cannot create a unit with a scale of 0.
+     ;;
+     ;; It might happen due to older version of NumPy in Guix, upastream
+     ;; tested with numpy==2.1.2, where we have 1.23.2.
+     (list #:tests? #f))
+    (native-inputs
+     (list python-pytest-astropy
+           python-setuptools
+           python-setuptools-scm
+           python-wheel))
     (propagated-inputs
      (list python-astropy
            python-matplotlib
            python-numpy
            python-scipy
            python-six))
-    (native-inputs (list python-pytest-astropy python-setuptools-scm))
     (home-page "https://radio-beam.readthedocs.io/en/latest/")
     (synopsis "Operations for radio astronomy beams with Astropy")
     (description
-- 
2.46.0

